src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -1681,6 +1331,457 @@ DebugIpStatusCollector::populateILAResults(boost::property_tree::ptree &_pt)
 
 }
 
+// ----- Supporting Functions -------------------------------------------
+
+void
+reportOverview(std::ostream& _output, const boost::property_tree::ptree& _dbgIpStatus_pt)
+{
+  _output << "Debug IP Status" << std::endl ;
+  _output << "  Number of IPs found :: " << _dbgIpStatus_pt.get<uint64_t>("total_num_debug_ips") << std::endl; // Total count with the IPs actually shown
```

Comment:
std::endl should not be used unless stream flushing is required. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rio-endl